app/controllers/polls_controller.rb

```diff
@@ -27,12 +27,19 @@ def show
   # GET /polls/new
   # GET /polls/new.json
   def new
-    @poll = Poll.new
+    if !current_user.admin?
```

Comment:
Instead of manually checking if a user is an admin, why not add the following line to the top of this file:

```ruby
  load_and_authorize_resource only: [:new, :create, :edit, :update]
```

That will not only redirect if the user isn't allowed to create a poll, but it will also set the `@poll` object for the `create`, `edit`, and `update` actions.